lib/validator.js

```diff
@@ -293,25 +295,26 @@ function checkSubSchema(schema, parent, state) {
     }
     if (typeof schema.xml !== 'undefined') {
         schema.xml.should.be.an.Object();
+        schema.xml.should.not.be.an.Array();
     }
     // example can be any type
 
     if (schema.externalDocs) {
         schema.externalDocs.should.have.key('url');
         should.doesNotThrow(function() {
- validateUrl(schema.externalDocs.url, [state.openapi.servers], 'externalDocs', state.options)
-}, 'Invalid externalDocs.url');
+            const servers = (state.openapi.servers === 'undefined') ? state.openapi.servers : [];
```

Comment:
Testing `state.openapi.servers` against the string `undefined` isn't going to work as expected. I think we're missing a `typeof` and a `!==` here? Also I *think* the `contextServers` passed into validateUrl is an **array** of server arrays in the order they should be tested, i.e. it could be link-level `servers` then operation-level `servers` then the main openapi-level `servers` all wrapped up into an array.

Reply:
Thank you! That ternary was a mistake, and a silly one. The array of arrays thing was definitely something I was concerned about, so thanks for helping me out there! 

I've made the changes, could you check they're ok?